src/fides/api/ops/util/data_category.py

```diff
@@ -20,8 +49,14 @@ def generate_fides_data_categories() -> Type[EnumType]:
 
 def _validate_data_category(data_category: str) -> str:
     """Checks that the data category passed in is currently supported."""
-    valid_categories = DataCategory.__members__.keys()
-    if data_category not in valid_categories:
+    from_db = [cat.fides_key for cat in get_fides_data_categories_from_db()]
+    from_taxonomy = [cat.fides_key for cat in get_fides_data_categories_from_taxonomy()]
+    if all(
+        [
+            data_category not in from_db,
+            data_category not in from_taxonomy,
+        ]
+    ):
         raise common_exceptions.DataCategoryNotSupported(
             f"The data category {data_category} is not supported."
         )
```

Comment:
We added similar logic in the Unified Fides Resources release - can we consolidate please?   We have too many "validate data category" methods.

https://github.com/ethyca/fides/blob/main/src/fides/api/ops/api/v1/endpoints/dataset_endpoints.py#L87-L99